test/integration/local/test_mnist.py

```diff
@@ -12,113 +12,161 @@
+    if instance_type == 'local_gpu':
+        pytest.skip('Local Mode does not support distributed GPU training.')
 
-    cluster_size = 2
     # pure_nccl communicator hangs when only one gpu is available.
+    cluster_size = 2
     hyperparameters = {'sagemaker_process_slots_per_host': 1,
                        'sagemaker_num_processes': cluster_size,
                        'batch-size': 10000,
                        'epochs': 1,
                        'communicator': 'hierarchical'}
 
-    local_mode.train(customer_script, data_dir, docker_image, opt_ml,
-                     hyperparameters=hyperparameters,
-                     cluster_size=cluster_size, source_dir=mnist_path, use_gpu=use_gpu)
+    estimator = Chainer(entry_point=customer_script,
+                        source_dir=mnist_path,
+                        role='SageMakerRole',
+                        image_name=docker_image,
+                        train_instance_count=cluster_size,
+                        train_instance_type=instance_type,
+                        sagemaker_session=sagemaker_local_session,
+                        hyperparameters=hyperparameters,
+                        output_path='file://{}'.format(tmpdir))
+
+    estimator.fit({'train': 'file://{}'.format(os.path.join(data_dir, 'train')),
+                   'test': 'file://{}'.format(os.path.join(data_dir, 'test'))})
+
+    success_files = {
+        'model': ['model.npz'],
+        'output': ['success', 'data/accuracy.png', 'data/cg.dot', 'data/log', 'data/loss.png'],
+    }
+
+    test_utils.files_exist(str(tmpdir), success_files)
+
+    request_data = np.zeros((100, 784), dtype='float32')
+
+    test_utils.predict_and_assert_response_length(estimator, request_data, instance_type)
+    test_utils.predict_and_assert_response_length(estimator, request_data, instance_type,
+                                                  json_serializer, json_deserializer,
+                                                  'application/json')
+    test_utils.predict_and_assert_response_length(estimator, request_data, instance_type,
+                                                  csv_serializer, csv_deserializer, 'text/csv')
 
-    files = ['model/model.npz', 'output/success', 'output/data/accuracy.png',
-             'output/data/cg.dot', 'output/data/log', 'output/data/loss.png']
 
-    test_utils.files_exist(opt_ml, files)
+def _json_predictor(estimator, instance_type):
+    predictor = estimator.deploy(1, instance_type)
+    predictor.content_type = 'application/json'
+    predictor.serializer = json_serializer
+    predictor.accept = 'application/json'
+    predictor.deserializer = json_deserializer
+    return predictor
 
-    assert not local_mode.file_exists(opt_ml, 'output/failure'), 'Failure happened'
 
-    with local_mode.serve(os.path.join(mnist_path, customer_script), model_dir=None,
-                          image_name=docker_image, opt_ml=opt_ml):
+class CsvDeserializer():
```

Comment:
Can we add this to the python sdk?

Reply:
sure - https://github.com/aws/sagemaker-python-sdk/pull/737